### PR TITLE
[ServiceBus/EventHub] add service specific message annotations to receiver logs

### PR DESCRIPTION
**Pull request by swathipil**
Adding service specific annotations/other metadata on received messages to help with debugging. Sample log with addition:

`2024-11-01 08:25:11,774 | MainThread | DEBUG | azure.servicebus._transport._pyamqp_transport | Received message: seq-num: 11247, enqd-utc: datetime.datetime(2024, 11, 1, 15, 25, 10, 802000, tzinfo=datetime.timezone.utc), lockd-til-utc: datetime.datetime(2024, 11, 1, 15, 30, 11, 5000, tzinfo=datetime.timezone.utc), ttl: None, dlvry-cnt: 0`.

Larger snippet:
```
2024-11-01 08:25:11,706 | MainThread | DEBUG | azure.servicebus._pyamqp.session | -> FlowFrame(next_incoming_id=3, incoming_window=65534, next_outgoing_id=1, outgoing_window=65535, handle=3, delivery_count=1, link_credit=1, available=None, drain=None, echo=None, properties=None)
2024-11-01 08:25:11,773 | MainThread | DEBUG | azure.servicebus._pyamqp.receiver | <- TransferFrame(handle=2, delivery_id=2, delivery_tag=b'\x13\xca&!\xf8Z\xb9F\xa0\xc4?\xfe\xac\xa4a\xb6', message_format=0, settled=None, more=False, rcv_settle_mode=None, state=None, resume=None, aborted=None, batchable=True, payload=b'***')
2024-11-01 08:25:11,774 | MainThread | DEBUG | azure.servicebus._transport._pyamqp_transport | Received message: seq-num: 11247, enqd-utc: datetime.datetime(2024, 11, 1, 15, 25, 10, 802000, tzinfo=datetime.timezone.utc), lockd-til-utc: datetime.datetime(2024, 11, 1, 15, 30, 11, 5000, tzinfo=datetime.timezone.utc), ttl: None, dlvry-cnt: 0
Message of sequence body received. Body is:
[b'message', 123.456, True]
2024-11-01 08:25:11,775 | MainThread | DEBUG | azure.servicebus._pyamqp.receiver | -> DispositionFrame(role=True, first=2, last=None, settled=True, state=Accepted(), batchable=None)
2024-11-01 08:25:11,775 | MainThread | DEBUG | azure.servicebus._pyamqp.cbs | CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
2024-11-01 08:25:11,775 | MainThread | DEBUG | azure.servicebus._pyamqp.session | -> FlowFrame(next_incoming_id=4, incoming_window=65533, next_outgoing_id=1, outgoing_window=65535, handle=3, delivery_count=2, link_credit=1, available=None, drain=None, echo=None, properties=None)
```

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py [hunk 1]
```diff
@@ -78,6 +78,12 @@ async def _incoming_transfer(self, frame):
                 self._received_payload = bytearray()
             else:
                 message = decode_payload(frame[11])
+            _LOGGER.debug(
+                "Received message: annotations: %r, header: %r",
+                message.message_annotations,
+                message.header,
```
Comment (swathipil) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py:84]: I don't think adding header here should expose any private info. I think it's useful to see delivery_count/ttl that may  exist in the header in some cases. Thoughts/concerns? Are there any scenarios I'm missing where this does expose info that we don't want to?

Comment (l0lawrence) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py:84]: I don't think so, one thing if we are exposing the locked_until_time, could we also log it when it gets decoded into utc time so its more human legible?

**Review by kashifkhan — APPROVED** [review]

File: sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py (hunk 1 above)
Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py:84]: sounds good . OOB: do if logging_enable, output only values where key is x-opt- (i.e. service vals, so we don't leak customer props) + rest.

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py:84]: if logging.getLogger().isEnabledFor(logging.DEBUG):

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_receiver_async.py]: updated!

**Pull request merged**